src/admin.rs

```diff
@@ -182,10 +180,6 @@ pub fn load_aws() -> Result<Arc<dyn ObjectStore>, Box<dyn Error>> {
     let builder = if let Some(dynamodb_table) = dynamodb_table {
         builder.with_conditional_put(S3ConditionalPut::Dynamo(DynamoCommit::new(dynamodb_table)))
     } else {
```

Comment:
should we be setting `with_conditional_put` in this branch? I think object_store will return an error when using `PutMode.Create` if this is not set

Reply:
Ahhh, good call. Using ETagMatch, right?

https://docs.rs/object_store/latest/object_store/aws/enum.S3ConditionalPut.html

cc @benesch that's the intended use, isn't it?

Reply:
Ah yep, that's right!

If you need [`S3CopyIfNotExists`](https://docs.rs/object_store/latest/object_store/aws/enum.S3CopyIfNotExists.html) the story is a bit more complicated, but I don't think SlateDB currently uses that?

Reply:
`ETagPutIfNotExists` is what we're after, I think.

Reply:
Ah, I see it was removed in favor of ETagMatch. :) Updating.

Reply:
Yep! `ETagPutIfNotExists` was a short lived variant for the brief window where S3 supported `If-None-Match` but not `If-Match`. It never made it into a release of `object_store`